### PR TITLE
Baizhu snakey implementation

### PR DESCRIPTION
**Pull request by IvanHdzF**
I sincerely hope nothing broke

### CODE REVIEW

File: internal/characters/baizhu/asc.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+}
+func (c *char) a4() {
+	if c.Base.Ascension < 4 {
+		return
+	}
+	c.Core.Player.ActiveChar().AddReactBonusMod(character.ReactBonusMod{
+		Base: modifier.NewBase("baizhu-a4", 6*60),
+		Amount: func(ai combat.AttackInfo) (float64, bool) {
+			limitHP := c.MaxHP() / 1000 //TODO:Is this an integer div or a float one in game?
```
Comment (unleashurgeek) [inline internal/characters/baizhu/asc.go:56]: float

File: internal/characters/baizhu/attack.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+	"github.com/genshinsim/gcsim/pkg/core/geometry"
+)
+
+const normalHitNum = 4
+
+var (
+	attackFrames   [][]int
+	attackHitmarks = [][]int{{12}, {13}, {21, 26}, {13}}
+	attackHitboxes = [][]float64{{2, 3}, {2, 3}, {2.2}, {2, 3}}
+	attackOffsets  = []float64{-0.2, -0.2, 1.1, -0.2}
```
Comment (unleashurgeek) [inline internal/characters/baizhu/attack.go:20]: Fix last 2 hitboxes (all boxes, no circles):
3. width = `2.4`, height = `3.0`, offset = `-0.2`
4. width = `3.2`, height = `3.0`, offset = `-0.2`

File: internal/characters/baizhu/cons.go [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+// Universal Diagnosis gains 1 additional charge.
+func (c *char) c1() {
+	c.SetNumCharges(action.ActionSkill, 2)
+}
+
+// When your own active character hits a nearby opponent with their attacks, Baizhu will unleash a Gossamer Sprite: Splice.
+// Gossamer Sprite: Splice will initiate 1 attack before returning, dealing 300% of Baizhu's ATK as Dendro DMG and healing for
+// 20% of Universal Diagnosis's Gossamer Sprite's normal healing.
+// DMG dealt this way is considered Elemental Skill DMG.
+// This effect can be triggered once every 6s.
```
Comment (unleashurgeek) [inline internal/characters/baizhu/cons.go:25]: fix: [Incisive Discernment](https://genshin.honeyhunterworld.com/c_822/?lang=EN)
```
When your own active character hits a nearby opponent with their attacks, Baizhu will unleash a Gossamer Sprite: Splice.
Gossamer Sprite: Splice will initiate 1 attack before returning, dealing 250% of Baizhu's ATK as Dendro DMG and healing for 20% of Universal Diagnosis's Gossamer Sprite's normal healing.
DMG dealt this way is considered Elemental Skill DMG.
This effect can be triggered once every 5s.
```

File: internal/characters/baizhu/cons.go [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+				return m, true
+			},
+		})
+	}
+
+}
+
+// Increases the DMG dealt by Healing Holism's Spiritveins by 6% of Baizhu's Max HP.
+// Additionally, when Gossamer Sprite or Gossamer Sprite: Splice hit opponents, there is a 100% chance of generating one of Healing Holism's
+// Seamless Shields. This effect can only be triggered once by a Gossamer Sprite or Gossamer Sprite: Splice.
```
Comment (unleashurgeek) [inline internal/characters/baizhu/cons.go:96]: fix comment + implementaiton [Elimination of Malicious Qi](https://genshin.honeyhunterworld.com/c_826/?lang=EN)

```
Increases the DMG dealt by Holistic Revivification's Spiritveins by 8% of Baizhu's Max HP.
```

File: internal/characters/baizhu/skill.go [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+		}
+		//queue an attack vs next target
+		atk := *c.skillAtk
+		atk.SourceFrame = src
+		atk.Pattern = combat.NewSingleTargetHit(next.Key())
+		cb := c.chain(src, count+1)
+		if cb != nil {
+			atk.Callbacks = append(atk.Callbacks, cb)
+		}
+		c.Core.QueueAttackEvent(&atk, 10) //TODO: Modify delay
```
Comment (unleashurgeek) [inline internal/characters/baizhu/skill.go:99]: want to say 1s based on gameplay and config but not 100% sure

File: internal/characters/baizhu/burst.go [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+		AttackTag:  attacks.AttackTagElementalBurst,
+		ICDTag:     attacks.ICDTagNone,
+		ICDGroup:   attacks.ICDGroupDefault,
+		StrikeType: attacks.StrikeTypeDefault,
+		Element:    attributes.Dendro,
+		Durability: 25,
+		Mult:       burstAtk[c.TalentLvlBurst()],
+	}
+	if c.Base.Cons >= 6 {
+		ai.FlatDmg = c.MaxHP() * 0.06
```
Comment (unleashurgeek) [inline internal/characters/baizhu/burst.go:81]: `0.08`

File: internal/characters/baizhu/burst.go [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+	}
+
+	// TODO: Hitbox and strike timing
+	c.Core.QueueAttack(
+		ai,
+		combat.NewBoxHitOnTarget(c.Core.Combat.Player(), nil, 4.5, 8),
+		0,
+		11,
+	)
```
Comment (unleashurgeek) [inline internal/characters/baizhu/burst.go:90]: radius = 1.5 (can have delay match fischl projectile or something)

File: internal/characters/baizhu/skill.go (hunk 5 above)
Comment (IvanHdzF) [reply internal/characters/baizhu/skill.go:99]: I will wait for the frame counter speedrunner for this

**Review by unleashurgeek — APPROVED** [review]

**Pull request merged**